lib/screens/translation_screen.dart

```diff
@@ -68,30 +73,27 @@ class TranslationScreenState extends State<TranslationScreen> {
     if (_currentSentence == null) {
       _currentSentence = _sentences.first;
     }
-    String fileId = _currentSentence.id;
-    new File("$recordStorage/$fileId.$fileExtension").create().then((filePath) {
-      _flutterSound.startRecorder(filePath.path).then((path) {
-        setState(() {
-          this._isRecording = true;
-        });
-
-        _recorderSubscription =
-            _flutterSound.onRecorderStateChanged.listen((e) {
-          if (e == null) {
-            return;
-          }
-          DateTime date = new DateTime.fromMillisecondsSinceEpoch(
-              e.currentPosition.toInt());
-          String txt = DateFormat('mm:ss:SS', 'en_US').format(date);
+    String filePath = "$appDirectory/${_currentSentence.id}.$fileExtension";
+    _flutterSound.startRecorder(filePath).then((path) {
+      setState(() {
+        this._isRecording = true;
+      });
+      print(path);
+      _recorderSubscription = _flutterSound.onRecorderStateChanged.listen((e) {
+        if (e == null) {
+          return;
+        }
+        DateTime date =
+            new DateTime.fromMillisecondsSinceEpoch(e.currentPosition.toInt());
+        String txt = DateFormat('mm:ss:SS', 'en_US').format(date);
```

Comment:
Check if formating does not already provide the final string. So you don't have to do substring afterwards.

Reply:
No the formatting does not provide the final string. This is an example of what you get when formatting "00:05:437"